src/privileged/autoplay/api.js

```diff
@@ -0,0 +1,397 @@
+"use strict";
+
+ChromeUtils.import("resource:///modules/PermissionUI.jsm");
+ChromeUtils.import("resource:///modules/SitePermissions.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionStorage.jsm");
+ChromeUtils.import("resource://gre/modules/PopupNotifications.jsm");
+ChromeUtils.import("resource://gre/modules/PrivateBrowsingUtils.jsm");
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/TelemetryController.jsm");
+
+const CID = ChromeUtils.import("resource://gre/modules/ClientID.jsm", {});
+const { AddonStudies } = ChromeUtils.import("resource://normandy/lib/AddonStudies.jsm", {});
+const { EventManager } = ExtensionCommon;
+
+XPCOMUtils.defineLazyGetter(this, "gBrowserBundle", function() {
+  return Services.strings.createBundle("chrome://browser/locale/browser.properties");
+});
+
+function _once(target, name) {
```

Comment:
Hm. I might be misunderstanding the intent here, but [`addEventListener` takes a boolean named `once`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) if you just want an event listener to fire a single time:

Reply:
Since I want to use await pattern [here](https://github.com/alastor0325/autoplay-shield-study/blob/995dd3286df3ae659985a899936ceda691b47849/src/privileged/autoplay/api.js#L370) for waiting until the event happen, `_once` is just a wrapper helper function.